R/pivot-long.R

```diff
@@ -208,6 +208,11 @@ pivot_longer_spec <- function(data, cols,
     names[[col]] <- vec_cast(names[[col]], names_ptypes[[col]])
   }
 
+  # test the value of `col when it` uses `matches()` arg
+  if (length(cols) == 0) {
+    abort(glue::glue("Val ue in `matches()` has no corresponding value in `data`."))
```

Comment:
I think this needs to be a bit more general because it could be the result of any tidyselect function failing to match. Maybe something like `` `cols` must select at least one column. ``?

Do you want to also have a go at adding a test?

Reply:
I gave it a try! Thanks for the feedback.